BusBoard.Web/Views/Home/BusInfo.cshtml

```diff
@@ -6,5 +6,39 @@
 <h2>BusInfo</h2>
 
 <div class="row">
-    <p>You entered postcode @Model.PostCode</p>
+    @{
+        if (string.IsNullOrEmpty(Model.PostCode))
+        {
+            <p>You didn't enter a postcode!</p>
+            return;
+        }
+
+        <p>You entered postcode: @Model.PostCode</p>
+
+        var postcodeEntry = BusBoard.Api.ApiHelper.GetPostcodeEntry(Model.PostCode);
```

Comment:
Please move all logic like this to the view model. In this file you should just be displaying information from the view model. This might require the odd if statement (e.g. to display one thing if the model has an error property, display something else otherwise) and the odd for loop (e.g. to display a list of items in the model). But it should be mostly html and any logic should live in the view model. Does that make sense? 🙂